### ISSUE TITLE
database dropdown changes unexpectedly when opening second SQL Script

### ISSUE THREAD

**Issue by Tomikoma** [Tomikoma/dbeaver2#26]
<a href="https://github.com/verajosemanuel"><img src="https://avatars1.githubusercontent.com/u/15836677?v=4" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [verajosemanuel](https://github.com/verajosemanuel)**
_Monday Nov 21, 2016 at 09:25 GMT_
_Originally opened as https://github.com/dbeaver/dbeaver/issues/987_

----

Using Mysql database with several databases in the same server, the database dropdown changes active database unexpectedly by itself when executing more than one script.

Steps to reproduce issue (Community Edition 3.7.8/3.8.0 Windows 7, java version "1.8.0_101", 32bits)

- open a script tab > select database from drop down > execute a simple select query
- open a second script tab > select a **different** database from drop down > Execute select

The dropdown database selector **changed to the first selected database in BOTH script tabs,** making subsequent sql sentences using the second script tab to fail.

"autosync connection with navigator" has no effect in order to correct this issue checked or not.

Worst of all is when databases has the same tables/fields. You can never be sure if recordset is from the desired database.

**Comment by Tomikoma** [comment]
<a href="https://github.com/serge-rider"><img src="https://avatars0.githubusercontent.com/u/6398845?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [serge-rider](https://github.com/serge-rider)**
_Wednesday Nov 23, 2016 at 12:21 GMT_

----

Currently active database change affects whole connection (you can see it in database navigator tree). So it will be changed for all connection editors.
So basically it is a feature, not a bug. But this may lead to problems, I agree.

Maybe it makes sense to have different active database for each editor. Although it is not obvious how this should be visualized in database navigator tree.

For now as a workaround you can create multiple connections (using copy-paste) and set different active databases in editors.

**Comment by Tomikoma** [comment]
<a href="https://github.com/tati-kru"><img src="https://avatars1.githubusercontent.com/u/30336552?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [tati-kru](https://github.com/tati-kru)**
_Sunday Sep 17, 2017 at 16:48 GMT_

----

Hello @verajosemanuel,
Does this problem continue to be actual or workaround works fine?

**Comment by Tomikoma** [comment]
<a href="https://github.com/verajosemanuel"><img src="https://avatars1.githubusercontent.com/u/15836677?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [verajosemanuel](https://github.com/verajosemanuel)**
_Thursday Sep 21, 2017 at 10:11 GMT_

----

workaround is not a viable solution since I have more than 100 databases per server. Creating a separate connection per database is pain
